pgjdbc/src/main/java/org/postgresql/core/CommandCompleteParser.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2018, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.core;
+
+import org.postgresql.util.GT;
+import org.postgresql.util.PSQLException;
+import org.postgresql.util.PSQLState;
+
+/**
+ * Parses {@code oid} and {@code rows} from a {@code CommandComplete (B)} message (end of Execute).
+ */
+public class CommandCompleteParser {
+  private long oid;
+  private long rows;
+
+  public CommandCompleteParser() {
+  }
+
+  public long getOid() {
+    return oid;
+  }
+
+  public long getRows() {
+    return rows;
+  }
+
+  public void set(long oid, long rows) {
+    this.oid = oid;
+    this.rows = rows;
+  }
+
+  /**
+   * Parses {@code CommandComplete (B)} message.
+   * Status is in the format of "COMMAND OID ROWS" where both 'OID' and 'ROWS' are optional
+   * and COMMAND can have spaces within it, like CREATE TABLE.
+   *
+   * @param status COMMAND OID ROWS message
+   * @throws PSQLException in case the status cannot be parsed
+   */
+  public void parse(String status) throws PSQLException {
+    // Assumption: command neither starts nor ends with a digit
+    if (!Parser.isDigitAt(status, status.length() - 1)) {
+      set(0, 0);
+    }
+
+    // Scan backwards, while searching for a maximum of two number groups
+    //   COMMAND OID ROWS
+    //   COMMAND ROWS
+    long oid = 0;
+    long count = 0;
+    try {
+      int lastSpace = status.lastIndexOf(' ');
+      // Status ends with a digit => it is ROWS
+      if (Parser.isDigitAt(status, lastSpace + 1)) {
+        count = Parser.parseLong(status, lastSpace + 1, status.length());
+
+        if (Parser.isDigitAt(status, lastSpace - 1)) {
+          int penultimateSpace = status.lastIndexOf(' ', lastSpace - 1);
+          if (Parser.isDigitAt(status, penultimateSpace + 1)) {
+            oid = Parser.parseLong(status, penultimateSpace + 1, lastSpace);
+          }
+        }
+      }
+    } catch (NumberFormatException e) {
+      // As we're have performed a isDigit check prior to parsing, this should only
```

Comment:
As we have ?

Reply:
a isDigit -> isDigit?

Reply:
this comment isn't completely accurate. we have not done an isDigit call for each char we are parsing. We have definitely checked the first char and possibly have checked the last char.